src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -1253,3 +1254,101 @@ def get_parent_proc_name():
         parent_proc_name = _get_parent_proc_name()
         setattr(get_parent_proc_name, "return_value", parent_proc_name)
     return getattr(get_parent_proc_name, "return_value")
+
+
+def log_cmd_history(command, args):
+    import os
+    from knack.util import ensure_dir
+    from azure.cli.core.extension import get_extension, ExtensionNotInstalledException
+
+    if command == 'next':
+        return
+
+    # Determine whether "az next" has been installed.
+    # At present, command execution log is only recorded when "az next" is installed
+    try:
+        az_next_is_installed = get_extension("next")
+    except ExtensionNotInstalledException:
+        az_next_is_installed = False
+
+    if not az_next_is_installed:
+        return
+
+    # For commands that do not require logging, we can pass in the parameter "--no-log" to control
+    if '--no-log' in args:
+        return
```

Comment:
Is `--no-log` registered in `next` extension? It won't work without registration:

```
$ az group list --no-log
UnrecognizedArgumentError: unrecognized arguments: --no-log
```

Reply:
Yes, and `--no-log` is currently only used for internal logic in the `next` extension and will not have the same name as the existing parameters in CLI.